File: Data Structures & Algorithms/duplicate-integer/submission-2.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
        Set<Integer> distinct = new HashSet<>(
            Arrays.stream(nums).boxed().collect(Collectors.toList()));
        for (int n : nums) {
            if(!distinct.remove(n)) return true;
        }
        return false;
    }
}

File: Data Structures & Algorithms/length-of-last-word/submission-2.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        if len(s) == 1:
            return 1
        j = len(s)
        while s[j - 1] == " ":
            j -= 1
        i = j
        while s[i - 1] != " ":
            i -= 1
        return len(s[i : j])
            